### PR TITLE
ContentLinksResourceProcessor now supports projections

### LINKED ISSUE #513: Content links missing when using Spring Data REST projections
When using [projections](https://docs.spring.io/spring-data/rest/docs/current/reference/html/#projections-excerpts) with the `@Projection` annotation in Spring Data REST, the `content` association in the HATEAOS `_links` section is missing.

That is because Spring Content looks for fields annotated with `@ContentId` to generate the link in [ContentLinksResourceProcessor](https://github.com/paulcwarren/spring-content/blob/1.2.x/spring-content-rest/src/main/java/internal/org/springframework/content/rest/links/ContentLinksResourceProcessor.java).

But projections have no fields, only methods. Putting the `@ContentId` annotation on a `String getContentId()` method in the projection interface does not help.

I think this could be resolved by not only considering the fields, but also looking at the `propertyDescriptor.getReadMethod()` when checking if the `@ContentId` annotation is present.

Pseudo code:

```
var propertyDescriptor = new BeanWrapperImpl(object).getPropertyDescriptors()[index];
if (AnnotationUtils.findAnnotation(propertyDescriptor.getReadMethod(), ContentId.class) != null) {
    ...
}
```

@paulcwarren would you consider such a change ?

### PR DESCRIPTION
**Pull request by paulcwarren**
Fixes #513

### CODE REVIEW

File: spring-content-rest/src/main/java/internal/org/springframework/content/rest/links/ContentLinksResourceProcessor.java [hunk 1]
```diff
@@ -76,9 +76,10 @@ public PersistentEntityResource process(final PersistentEntityResource resource)
 
 		Object entityId = DomainObjectUtils.getId(object);
```
Comment (tgeens) [inline spring-content-rest/src/main/java/internal/org/springframework/content/rest/links/ContentLinksResourceProcessor.java:77]: Is that under the assumption that the projection actually exposes the actual id ? Or does that not matter ?

Comment (tgeens) [reply spring-content-rest/src/main/java/internal/org/springframework/content/rest/links/ContentLinksResourceProcessor.java:77]: Just FYI - I was using this in my own proof-of-concept (but that might be totally wrong):
```
Object entityId = DomainObjectUtils.getId(resource.getTarget());
```

Comment (paulcwarren) [reply spring-content-rest/src/main/java/internal/org/springframework/content/rest/links/ContentLinksResourceProcessor.java:77]: Yes, at the moment this assumes the projection exposes a method annotated with @Id that returns the id.  

I want to try and remove, if possible.

Comment (paulcwarren) [reply spring-content-rest/src/main/java/internal/org/springframework/content/rest/links/ContentLinksResourceProcessor.java:83]: I refactored this change a little bit and it now should handle projections correctly essentially the same as regular entities.  

Behind the scenes spring data is creating a proxy and with a bit of digging I discovered its possible to de-reference it to find the actual object.  Test proves you now don't need an annotated `getId` method.

You should be able to test this by grabbing the latest snapshots.  Available from `https://oss.sonatype.org/content/repositories/snapshots`.

**Pull request merged**